pkg/query/storeset.go

```diff
@@ -238,10 +239,12 @@ func (s *StoreSet) Update(ctx context.Context) {
 		// No external labels means strictly store gateway or ruler and it is fine to have access to multiple instances of them.
 		//
 		// Sidecar will error out if it will be configured with empty external labels.
-		if len(store.Labels()) > 0 && externalLabelStores[externalLabelsFromStore(store)] != 1 {
+		externalLabels := externalLabelsFromStore(store)
+		storesWithExternalLabels := externalLabelStores[externalLabels]
+		if len(store.LabelSets()) > 0 && len(store.LabelSets()[0].Labels) > 0 && storesWithExternalLabels != 1 {
 			store.close()
 			s.updateStoreStatus(store, errors.New(droppingStoreMessage))
-			level.Warn(s.logger).Log("msg", droppingStoreMessage, "address", addr)
+			level.Warn(s.logger).Log("msg", droppingStoreMessage, "address", addr, "externalLabels", externalLabels, "duplicate-amount", storesWithExternalLabels)
```

Comment:
```suggestion
			level.Warn(s.logger).Log("msg", droppingStoreMessage, "address", addr, "extLset", externalLabels, "duplicates", storesWithExternalLabels)
```
I feel like short names are fine + let's be consistent on camelCase, snakeCase

Reply:
Done.